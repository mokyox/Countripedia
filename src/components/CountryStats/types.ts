export interface Props {
  setSelectedCountry?: React.Dispatch<React.SetStateAction<string>>;
  selectedCountry: {
    name?: string;
    capital: string;
    flag?: string;
    population?: number;
  };
  country?: {
    name?: string;
    capital?: string;
    flag?: string;
    population?: number;
    languages?: [{ name: string }];
  };
}
