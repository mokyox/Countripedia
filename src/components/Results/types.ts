export interface Props {
  filteredCountries: string[];
  isLoaded: boolean;
  setSelectedCountry: React.Dispatch<React.SetStateAction<string>>;
  selectedCountry: string;
}
