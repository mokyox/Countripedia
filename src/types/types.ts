export interface CountryStatsProps {
  selectedCountry: {
    name: string;
    capital: string;
    flag: string;
    population: number;
    languages: [{ name: string }];
  };
}

export interface ResultsProps extends CountryStatsProps {
  filteredCountries: [{ name: string }];
  IsCountryLoaded: boolean;
  setSelectedCountry: React.Dispatch<React.SetStateAction<{ name: string }>>;
}
