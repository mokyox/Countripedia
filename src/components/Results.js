import React, { useState } from "react";
import CountryStats from "./CountryStats";
import PropTypes from "prop-types";
import styled from "styled-components";

const Results = ({ filteredCountries }) => {
  const [selectedCountry, setSelectedCountry] = useState("");
  const [isClicked, setIsClicked] = useState(false);
  console.log(selectedCountry.name, "is the selectedCountry");
  if (filteredCountries.length > 10) {
    return <p>Too many matches, please specify another filter.</p>;
  }
  return (
    <>
      <StyledResults>
        <ul>
          {filteredCountries.map(country => (
            <React.Fragment key={country.name}>
              <StyledResult>
                <li>{country.name}</li>
                <StyledButton
                  onClick={() => {
                    setSelectedCountry(country);
                    setIsClicked(!isClicked);
                  }}
                >
                  Display
                </StyledButton>
              </StyledResult>
            </React.Fragment>
          ))}
        </ul>
      </StyledResults>
      {selectedCountry ? <CountryStats country={selectedCountry}></CountryStats> : ""}
    </>
  );
};

const StyledResults = styled.div`
  /* border: solid 3px green; */
  text-align: center;
  margin: 1rem 0;
`;

const StyledResult = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: nowrap;
`;

const StyledButton = styled.button`
  background: #342d28;
  border: 1px solid #342d28;
  color: white;
  padding: 0.3rem 0.5rem;
  border-radius: 5px;
  text-decoration: none;
  font-weight: 700;
  transition: all 0.1s ease-in-out;
  &:hover {
    transform: scale(1.1);
  }
  &:hover,
  &:focus {
    outline: none;
  }
  margin: 0 20px;
`;

Results.propTypes = {
  filteredCountries: PropTypes.array
};

export default Results;
