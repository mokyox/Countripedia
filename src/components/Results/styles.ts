import styled from "styled-components";

export const StyledResults = styled.div`
  margin: 0.5 rem auto;
  max-width: 480px;
  text-align: center;
  padding: 1rem;
`;

export const StyledResult = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

export const StyledButton = styled.button`
  background: #ac5610;
  border: 1px solid #ac5610;
  color: white;
  padding: 0.3rem 0.5rem;
  border-radius: 5px;
  text-decoration: none;
  font-weight: 700;
  transition: all 0.1s ease-in-out;
  margin: 5px 1rem;
  cursor: pointer;
  &:hover {
    transform: scale(1.1);
  }
  &:hover,
  &:focus {
    outline: none;
  }
`;
