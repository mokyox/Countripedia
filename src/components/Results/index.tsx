// @ts-nocheck

import React, { useState } from "react";
import CountryStats from "../CountryStats";
import { Props } from "./types";
import { StyledButton, StyledResult, StyledResults } from "./styles";

const Results: React.FC<Props> = ({
  filteredCountries,
  IsCountryLoaded,
  setSelectedCountry,
  selectedCountry
}) => {
  const [isClicked, setIsClicked] = useState(false);

  //Get API weather data after country API data from parent component is called

  if (filteredCountries.length > 10) {
    return <p> Please specify filter further.</p>;
  }
  if (filteredCountries.length === 0 && IsCountryLoaded) {
    return <p>No countries found.</p>;
  }
  return (
    <>
      <StyledResults>
        <ul>
          {filteredCountries.map(country => (
            <React.Fragment key={country.name}>
              <StyledResult>
                <li>{country.name}</li>
                <StyledButton
                  onClick={() => {
                    setSelectedCountry(country);
                    setIsClicked(!isClicked);
                  }}
                >
                  Display
                </StyledButton>
              </StyledResult>
            </React.Fragment>
          ))}
        </ul>
      </StyledResults>
      {selectedCountry ? (
        <CountryStats selectedCountry={selectedCountry}></CountryStats>
      ) : (
        ""
      )}
    </>
  );
};

export default Results;
