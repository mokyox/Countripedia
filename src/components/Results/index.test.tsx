// @ts-nocheck
//TODO: Fix TS errors with selectedCountry types
import React from "react";
import Results from "./";
import { render } from "@testing-library/react";
import { CountryStatsProps } from "../../types/types";

const filteredCountries = [
  { name: "Azerbaijan" },
  { name: "Jamaica" },
  { name: "Japan" },
  { name: "Svalbard and Jan Mayen" },
];

const selectedCountry: CountryStatsProps = {
  name: "Sweden",
  capital: "Stockholm",
  flag: "Swedish Flag",
  population: 9894888,
  languages: [{ name: "Swedish" }],
};

const setSelectedCountry = jest.fn();
const useStateSpy = jest.spyOn(React, "useState");
useStateSpy.mockImplementation((selectedCountry) => [
  selectedCountry,
  setSelectedCountry,
]);

describe("<Results />", () => {
  it("renders a <Results /> component", () => {
    const { getByTestId } = render(
      <Results
        filteredCountries={filteredCountries}
        selectedCountry={selectedCountry}
        setSelectedCountry={setSelectedCountry}
      />
    );
    expect(getByTestId("results")).toMatchSnapshot();
  });
});
