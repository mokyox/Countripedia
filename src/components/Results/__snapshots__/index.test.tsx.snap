// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> renders a <Results /> component 1`] = `
<div
  class="sc-AxiKw jMHJWW"
  data-testid="results"
>
  <ul>
    <div
      class="sc-AxhCb eDqXhQ"
    >
      <li>
        Azerbaijan
      </li>
      <button
        class="sc-AxhUy cuekrc"
      >
        Display
      </button>
    </div>
    <div
      class="sc-AxhCb eDqXhQ"
    >
      <li>
        Jamaica
      </li>
      <button
        class="sc-AxhUy cuekrc"
      >
        Display
      </button>
    </div>
    <div
      class="sc-AxhCb eDqXhQ"
    >
      <li>
        Japan
      </li>
      <button
        class="sc-AxhUy cuekrc"
      >
        Display
      </button>
    </div>
    <div
      class="sc-AxhCb eDqXhQ"
    >
      <li>
        Svalbard and Jan Mayen
      </li>
      <button
        class="sc-AxhUy cuekrc"
      >
        Display
      </button>
    </div>
  </ul>
</div>
`;
